frontend/src/OnBoardingForm/OnbboardingFromSH.jsx

```diff
@@ -0,0 +1,304 @@
+import React, { useState, useEffect } from 'react';
+import { authenticationService } from '@/_services';
+import { toast } from 'react-hot-toast';
+import { useHistory } from 'react-router-dom';
+import OnBoardingInput from './OnBoardingInput';
+import OnBoardingRadioInput from './OnBoardingRadioInput';
+import AdminSetup from './AdminSetup';
+import OnboardingBubblesSH from './OnboardingBubblesSH';
+import ContinueButtonSelfHost from './ContinueButtonSelfHost';
+
+function OnbboardingFromSH({ darkMode }) {
+  const history = useHistory();
+  const [page, setPage] = useState(0);
+  const [completed, setCompleted] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [formData, setFormData] = useState({
+    companyName: '',
+    role: '',
+    companySize: '',
+    name: '',
+    email: '',
+    password: '',
+    workspace: '',
+  });
+
+  const pageProps = {
+    formData,
+    setFormData,
+    setPage,
+    page,
+    setCompleted,
+    isLoading,
+    darkMode,
+  };
+
+  useEffect(() => {
+    if (completed) {
+      authenticationService
+        .setupAdmin({
+          companyName: formData.companyName.trim(),
+          companySize: formData.companySize,
+          role: formData.role,
+          password: formData.password,
+          name: formData?.name,
+          email: formData?.email,
+          workspace: formData?.workspace,
+        })
+        .then((user) => {
+          authenticationService.updateUser(user);
+          authenticationService.deleteLoginOrganizationId();
+          setIsLoading(false);
+          history.push('/');
+        })
+        .catch((res) => {
+          setIsLoading(false);
+          setCompleted(false);
+          toast.error(res.error || 'Something went wrong', {
+            id: 'toast-login-auth-error',
+            position: 'top-center',
+          });
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [completed]);
+
+  const FORM_TITLES = [
+    `Set up your admin account`,
+    `Set up your workspace`,
+    `Where do you work ${formData?.name}?`,
+    'What best describes your role?',
+    'What is the size of your company?',
+    'What is the size of your company?', //dummy placeholder
+  ];
+  const FormSubTitles = ['This information will help us improve ToolJet.'];
+
+  const getuserName = () => {
+    let nameArray = formData?.name?.split(' ');
+    if (nameArray?.length > 0)
+      return `${nameArray?.[0][0]}${nameArray?.[1] != undefined && nameArray?.[1] != '' ? nameArray?.[1][0] : ''} `;
+    return '';
+  };
```

Comment:
You can use this function as a helper function on both 
`frontend/src/OnBoardingForm/OnbboardingFromSH.jsx` and `frontend/src/OnBoardingForm/OnBoardingForm.jsx` pages